The original patch has been modified for aarch64. Upstream has already
integrated a better, more general fix, but this is simple and sufficient enough
to fix the compile problem in flathub.

From 1ea63ccb345a4498b41cb8842622ba7ecc9fd484 Mon Sep 17 00:00:00 2001
From: Gilles Caulier <caulier.gilles@gmail.com>
Date: Sun, 7 Jun 2020 10:37:13 +0200
Subject: [PATCH] fix broken compilation under ARM. tested with my Raspberry
 PIV and Raspbian 64bits.

---
 cmake/compilerFlags.cmake | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/cmake/compilerFlags.cmake b/cmake/compilerFlags.cmake
index aad1da17a..96cc0c323 100644
--- a/cmake/compilerFlags.cmake
+++ b/cmake/compilerFlags.cmake
@@ -26,7 +26,11 @@ if ( MINGW OR UNIX OR MSYS ) # MINGW, Linux, APPLE, CYGWIN
         # This fails under Fedora - MinGW - Gcc 8.3
         if (NOT MINGW)
             if (COMPILER_IS_GCC AND CMAKE_CXX_COMPILER_VERSION VERSION_GREATER_EQUAL 8.0)
-                add_compile_options(-fstack-clash-protection -fcf-protection)
+                if (NOT ${CMAKE_SYSTEM_PROCESSOR} MATCHES "aarch64")
+                    add_compile_options(-fstack-clash-protection -fcf-protection)
+                else()
+                    add_compile_options(-fstack-clash-protection)
+                endif()
             endif()
 
             if (COMPILER_IS_GCC OR (COMPILER_IS_CLANG AND CMAKE_CXX_COMPILER_VERSION VERSION_GREATER_EQUAL 3.7 ))
